decentraland/screens/BuyLandScreen.js

```diff
@@ -3,24 +3,28 @@ import { connect } from "react-redux";
 import { removeLandForSale } from "../redux/actions";
 import BuyLand from "@presentational/BuyLand";
 import PropTypes from "prop-types";
+import { showError, showInfo, getContracts } from "./helpers";
 
-import ContractsAddresses from "@constants/ContractsAddresses";
-const { ESTATE_ADDRESS, MARKETPLACE_ADDRESS } = ContractsAddresses;
+import AssetTypes from "@constants/AssetTypes";
+const { ESTATE, PARCEL } = AssetTypes;
 
-import { Action } from "tasit-sdk";
-const { ERC721, Marketplace } = Action;
-const { Estate } = ERC721;
-const { Decentraland: DecentralandMarketplace } = Marketplace;
-import { approveManaSpending, manaFaucetTo } from "./helpers";
+// TODO: Go deep on gas handling.
+// Without that, VM returns a revert error instead of out of gas error.
+// See: https://github.com/tasitlabs/TasitSDK/issues/173
+const gasParams = {
+  gasLimit: 7e6,
+  gasPrice: 1e9,
+};
 
 export class BuyLandScreen extends React.Component {
-  estateContract = new Estate(ESTATE_ADDRESS);
-  marketplaceContract = new DecentralandMarketplace(MARKETPLACE_ADDRESS);
-
   _onBuy = landForSale => {
-    const { account } = this.props;
-    if (!account) this._setupAccount();
-    else this._buy(landForSale);
+    try {
+      const { account } = this.props;
+      if (!account) this._setupAccount();
+      else this._buy(landForSale);
```

Comment:
Don't we want to buy it even if the account needs to get set up first? So maybe no `else` here?

Reply:
The current flow redirects the user back to the desired asset after account setup. The `_buy` function assumes that the account is created, funded and allowed. Any ideas to make that more clear?

Reply:
```suggestion
     // The _buy function assumes that the account is created, funded and allowed
     // The user will get back to this else case after going through an account setup flow
     else this._buy(landForSale);
```

Reply:
@marcelomorgado I committed a suggestion for this. Let me know if that works for you.